### DISCUSSION TITLE
Atmospheric Effects of Precipitation through Energetic X-Rays (AEPEX)

### DISCUSSION THREAD

**Discussion by git-addrian** [daniestevez/gr-satellites#673] [Satellite teams]
### Introduction

Our team at the Laboratory for Atmospheric and Space Physics (LASP) at the University of Colorado Boulder (CU) is preparing for the launch and commissioning of AEPEX in the near-future (2025). We would like to add gr-satellites support for AEPEX telemetry. 

LASP has previously created decoders for the following missions: [CUTE](https://github.com/daniestevez/gr-satellites/discussions/312), [CTIM](https://github.com/daniestevez/gr-satellites/discussions/355), [CIRBE](https://github.com/daniestevez/gr-satellites/discussions/423)

### AEPEX Mission

AEPEX is a 6U CubeSat that will study radiation belt losses through energetic particle precipitation (EPP) in the upper atmosphere, magnetopause shadowing, and wave-particle interactions. It uses a novel method to distinguish and quantify the impacts of these different components by measuring flux, spectrum, spatial scales, and temporal scales of EPP.

### Spacecraft Radios
| Band | UHF | S-Band|
|-------|------|---------|
|Type| SpaceQuest TRX-U| Clyde Space HSTX|
|Downlink Frequency| 437.325 MHz | 2405.0 MHz|
|Downlink Mode| GFSK|OQPSK|
|Downlink Baud|  9.6 kbaud and 19.2kbaud |  5 Mbaud and 10 Mbaud|
|Downlink Encoding| AX.25 -- G3RUH | CCSDS -- (Waiting on more details)|
| Uplink Frequency | 437.325 MHz | None|
|Uplink Mode| GFSK| None|
|Uplink Baud|  9.6 kbaud and 19.2kbaud |  None |
|Uplink Encoding| AX.25 -- G3RUH | None |

I have collected this information on the spacecraft radios. Please let me know if more info is needed!

### Decoder

In the next few days I will submit a pull request for the decoder I have built for AEPEX, from [this fork](https://github.com/lasp/gr-satellites) of gr-satellites. 
For testing purposes, I have uploaded the file "aepex_beacons.bin" within "aepex_beacons.zip", which can be decoded with the aepex_70cm.py decoder.
[aepex_beacons.zip](https://github.com/user-attachments/files/17499343/aepex_beacons.zip)

**Comment by K4KDR** [comment]
`the file "aepex_beacons.bin" within "aepex_beacons.zip", which can be decoded with the aepex_70cm.py decoder.`

Sorry to bother, but I've tried as many methods as I can think of but only get errors when I try this.   Could I trouble you for the syntax or process that one would use to decode that sample file with your python decoder?

Many thanks!

**Comment by daniestevez** [comment]
Thank you for opening a discussion for AEPEX!

I've taken a quick look at https://github.com/lasp/gr-satellites/commit/9e098d7c534b009f3ca94176456a0b4463e360c2 and this is the kind of thing that is needed to support the 70cm telemetry in gr-satellites: a SatYAML file and the telemetry definitions using construct. When the code is ready, send it as a pull request for review.

For the S-band radio, more information would be needed to support it, since CCSDS has many variants (some of which are already supported in gr-satellites). If you get more details, share them here. Also, an IQ recording would help.

**Reply by git-addrian** [reply]
Hey Daniel,
I have a pull request ready for you: https://github.com/daniestevez/gr-satellites/pull/675 

> For the S-band radio, more information would be needed to support it, since CCSDS has many variants (some of which are already supported in gr-satellites). If you get more details, share them here. Also, an IQ recording would help.

For now, it is likely that AEPEX will only be able to provide a 70cm decoder (similar to CUTE, CTIM, and CIRBE). As I track down more details on our specific variant of CCSDS, I can share them.
I can also see if I can find an IQ recording to supplement the binary file.

**Reply by daniestevez** [reply]
If you have details or recordings about the S-band downlink, please share them. Implementing a decoder for this is something that I might do.

**Comment by K4KDR** [comment]
Thank you both for all the work to get another LASP satellite into gr-satellites!

I fed the provided .BIN sample data file into a GNU Radio flowgraph using the newly added support for AEPEX...

![grc](https://github.com/user-attachments/assets/e6f0d5ae-d517-4aa8-adc4-a80a0035451a)

... and the output frame matches the source:

![grc-output_sample-bin](https://github.com/user-attachments/assets/d00f632d-b2cb-4ad9-9867-12cc10b439aa)

... I'll attach the **telemetry decoder output** here if you would like to review it to confirm that Amateur end-users such a myself are processing your data correctly!

[aepex_decoder_k4kdr.txt](https://github.com/user-attachments/files/17572837/aepex_decoder_k4kdr.txt)

**Reply by git-addrian** [reply]
Hi @K4KDR,

Nice! This exactly matches my output. Thank you for running that through a flowgraph.
